tml_decoder/models/guides/soft_prompt_guide.py

```diff
@@ -0,0 +1,79 @@
+from typing import List
+
+from numpy import ndarray
+from tml_decoder.encoders.abstract_encoder import AbstractEncoder
+from tml_decoder.generators.abstract_generator import AbstractGenerator
+from tml_decoder.models.guides.abstract_guide import AbstractGuide
+from torch import nn, optim
+from torch.optim.lr_scheduler import CosineAnnealingLR
+import torch
+
+DEFAULT_DEVICE = None
+
+if torch.cuda.is_available():
+    DEFAULT_DEVICE = "cuda"
+elif torch.backends.mps.is_available():
+    DEFAULT_DEVICE = "mps"
+else:
+    DEFAULT_DEVICE = "cpu"
```

Comment:
This occurs in 4 files in our repository. What would you say to ecapsulate it as a separate function?
```
def get_default_device_base_on_available_device():
    if torch.cuda.is_available():
        DEFAULT_DEVICE = "cuda"
    elif torch.backends.mps.is_available():
        DEFAULT_DEVICE = "mps"
    else:
      DEFAULT_DEVICE = "CPU"
```

Reply:
Ok